aws/rust-runtime/aws-config/src/provider_config.rs

```diff
@@ -266,7 +266,7 @@ impl ProviderConfig {
     /// Override the HTTPS connector for this configuration
     ///
     /// **Warning**: Use of this method will prevent you from taking advantage of the timeout machinery.
-    /// Consider `with_tcp_connector`.
+    /// Consider [`ProviderConfig::with_tcp_connector`].
```

Comment:
Will this prevent users from taking advantage of all timeouts or just the HTTP-specific ones?

Also, perhaps we could note this as a possible way to implement non-standard timeout mechanisms?

Reply:
just the HTTP ones